### ISSUE TITLE
module 'faiss' has no attribute 'swigfaiss'

### ISSUE THREAD

**Issue by njanakiev** [criteo/autofaiss#112]
```
python 3.8.12
autofaiss                 2.13.2                   pypi_0    pypi
faiss-cpu                 1.7.2                    pypi_0    pypi
libfaiss                  1.7.2            h2bc3f7f_0_cpu    pytorch
```

First of all, thank you for the great project! I get the error: `module 'faiss' has no attribute 'swigfaiss'` when running the following command:

```python
import autofaiss

autofaiss.build_index(
    "embeddings.npy",
    "autofaiss.index",
    "autofaiss.json",
    metric_type="ip",
    should_be_memory_mappable=True,
    make_direct_map=True)
```

The error appears when running it for `make_direct_map=True`.

Tested using `conda 4.11.0` or `mamba 0.15.3` using `pytorch` or `conda-forge` channel.

**Comment by rom1504** [comment]
Do you have the same error when using a venv instead of conda?

**Comment by njanakiev** [comment]
From my experience I have to rely more on conda/mamba with conda-forge/pytorch because of pytorch and all the other packages that are more difficult or sometimes even impossible to install without breaking the environment.

I found the following workaround to create the faiss direct map (tested with IVF and HNSW indices):

```python
ivf_index = faiss.try_extract_index_ivf(index)
if ivf_index:
    ivf_index.make_direct_map()

if hasattr(index, "make_direct_map"):
    index.make_direct_map()
```

Perhaps that is more general than using the `faiss.swigfaiss` module.

**Comment by yopknopixx** [comment]
I had the exact same issue. Turns out it was due to both `faiss-cpu` and `faiss-gpu` being installed.

**Comment by krasserm** [comment]
With faiss-cpu 1.7.3 (installed via pypi) this issue comes up again and can be reproduced with e.g.

```
embeddings = np.float32(np.random.rand(100, 512))
build_index(embeddings, make_direct_map=True, save_on_disk=False)
```

> AttributeError: module 'faiss' has no attribute 'swigfaiss'

With faiss-cpu 1.7.2 installed it works as expected.

**Comment by cwfparsonson** [comment]
FYI I had to avoid using conda and faiss-gpu

`conda uninstall faiss-gpu`
`pip install faiss-cpu==1.7.2`

**Comment by fvanlitsenburg** [comment]
Thought I'd quickly flag this to the devs, but running the latest version of Haystack's CPU version with YML with docker-compose also causes this error to pop up when building the image. By adding `pip install faiss-cpu==1.7.2` to the Dockerfile I circumvented it.

**Comment by BwandoWando** [comment]
> Thought I'd quickly flag this to the devs, but running the latest version of Haystack's CPU version with YML with docker-compose also causes this error to pop up when building the image. By adding `pip install faiss-cpu==1.7.2` to the Dockerfile I circumvented it.

I just encountered this minutes ago with my docker image and adding `pip install faiss-cpu==1.7.2` to the Docker file fixed it.

**Comment by victor-paltz** [comment]
Hello!
Is it happening with the latest version of autofaiss?
We corrected this issue in `autofaiss>=2.15.7` (see: https://github.com/criteo/autofaiss/commit/afb3c398444f5ab5c69bc61e71d56e591372be1d)
The workaround was indeed to use `faiss-cpu==1.7.2` back in the days.

**Comment by krasserm** [comment]
@victor-paltz, it works now, thanks for fixing!
